sklearn/mixture/tests/test_gaussian_mixture.py

```diff
@@ -0,0 +1,703 @@
+                             'or check for degenerate data.'
+                             % max_iter, g.fit, X)
+
+
+def test_gaussian_mixture_n_parameters():
+    # Test that the right number of parameters is estimated
+    rng = np.random.RandomState(0)
+    n_samples, n_features, n_components = 50, 5, 2
+    X = rng.randn(n_samples, n_features)
+    n_params = {'spherical': 13, 'diag': 21, 'tied': 26, 'full': 41}
+    for cv_type in COVARIANCE_TYPE:
+        g = GaussianMixture(
+            n_components=n_components, covariance_type=cv_type,
+            random_state=rng).fit(X)
+        assert_equal(g._n_parameters(), n_params[cv_type])
+
+
+def test_gaussian_mixture_aic_bic():
+    # Test the aic and bic criteria
+    rng = np.random.RandomState(0)
+    n_samples, n_features, n_components = 50, 3, 2
+    X = rng.randn(n_samples, n_features)
+    # standard gaussian entropy
+    sgh = 0.5 * (fast_logdet(np.cov(X.T, bias=1)) +
+                 n_features * (1 + np.log(2 * np.pi)))
+    for cv_type in COVARIANCE_TYPE:
+        g = GaussianMixture(
+            n_components=n_components, covariance_type=cv_type,
+            random_state=rng)
+        g.fit(X)
+        aic = 2 * n_samples * sgh + 2 * g._n_parameters()
+        bic = (2 * n_samples * sgh +
+               np.log(n_samples) * g._n_parameters())
+        bound = n_features / np.sqrt(n_samples)
+        assert_true((g.aic(X) - aic) / n_samples < bound)
+        assert_true((g.bic(X) - bic) / n_samples < bound)
+
+
+def test_gaussian_mixture_verbose():
+    rng = np.random.RandomState(0)
+    rand_data = RandomData(rng)
+    n_components = rand_data.n_components
+    for cov_type in COVARIANCE_TYPE:
+        X = rand_data.X[cov_type]
+        g = GaussianMixture(n_components=n_components, n_init=1,
+                            max_iter=100, reg_covar=0, random_state=rng,
+                            covariance_type=cov_type, verbose=1)
+        h = GaussianMixture(n_components=n_components, n_init=1,
+                            max_iter=100, reg_covar=0, random_state=rng,
+                            covariance_type=cov_type, verbose=2)
+        old_stdout = sys.stdout
+        sys.stdout = StringIO()
+        try:
+            g.fit(X)
+            h.fit(X)
+        finally:
+            sys.stdout = old_stdout
+
+
+def test_warm_start():
+    random_state = 0
+    rng = np.random.RandomState(random_state)
+    n_samples, n_features, n_components = 500, 2, 2
+    X = rng.rand(n_samples, n_features)
+
+    # Assert the warm_start give the same result for the same number of iter
+    g = GaussianMixture(n_components=n_components, n_init=1,
+                        max_iter=2, reg_covar=0, random_state=random_state,
+                        warm_start=False)
+    g.fit(X)
+    h = GaussianMixture(n_components=n_components, n_init=1,
+                        max_iter=1, reg_covar=0, random_state=random_state,
+                        warm_start=True)
+    h.fit(X).fit(X)
```

Comment:
Indeed you are right, I was mistaken. ElasticNet / Lasso follow the same pattern.